onlineapi/src/main/kotlin/com/codingblocks/onlineapi/models/RESTAPIModel.kt

```diff
@@ -0,0 +1,348 @@
+        val id: Int? = null,
+
+        @field:SerializedName("bumped_at")
+        val bumpedAt: String? = null,
+
+        @field:SerializedName("slug")
+        val slug: String? = null,
+
+        @field:SerializedName("views")
+        val views: Int? = null,
+
+        @field:SerializedName("last_posted_at")
+        val lastPostedAt: Any? = null,
+
+        @field:SerializedName("visible")
+        val visible: Boolean? = null,
+
+        @field:SerializedName("like_count")
+        val likeCount: Int? = null,
+
+        @field:SerializedName("image_url")
+        val imageUrl: Any? = null,
+
+        @field:SerializedName("bookmarked")
+        val bookmarked: Any? = null,
+
+        @field:SerializedName("last_poster_username")
+        val lastPosterUsername: String? = null,
+
+        @field:SerializedName("posters")
+        val posters: List<PostersItem?>? = null,
+
+        @field:SerializedName("pinned_globally")
+        val pinnedGlobally: Boolean? = null,
+
+        @field:SerializedName("reply_count")
+        val replyCount: Int? = null,
+
+        @field:SerializedName("tags")
+        val tags: List<Any?>? = null,
+
+        @field:SerializedName("archetype")
+        val archetype: String? = null,
+
+        @field:SerializedName("highest_post_number")
+        val highestPostNumber: Int? = null,
+
+        @field:SerializedName("closed")
+        val closed: Boolean? = null,
+
+        @field:SerializedName("unseen")
+        val unseen: Boolean? = null,
+
+        @field:SerializedName("posts_count")
+        val postsCount: Int? = null,
+
+        @field:SerializedName("excerpt")
+        val excerpt: Any? = null
+)
+
+data class PostersItem(
+
+        @field:SerializedName("primary_group_id")
+        val primaryGroupId: Any? = null,
+
+        @field:SerializedName("user_id")
+        val userId: Int? = null,
+
+        @field:SerializedName("extras")
+        val extras: Any? = null,
+
+        @field:SerializedName("description")
+        val description: String? = null
+)
+
+data class RatingModel(
+        val rating: String,
+        val count: Int,
+        val stats: List<Double>,
+        val userScore: Any?
+)
+
+class Leaderboard(@JvmField
+                  var userName: String) : BaseModel() {
+
+    @JvmField
+    var collegeName: String? = null
+    @JvmField
+    var photo: String? = null
+    @JvmField
+    var score: Int? = 0
+}
+
+data class PostStream(
+        val unpinned: Any? = null,
+        val hasDeleted: Boolean? = null,
+        val pinned: Boolean? = null,
+        val chunkSize: Int? = null,
+        val suggestedTopics: List<SuggestedTopicsItem?>? = null,
+        val currentPostNumber: Int? = null,
+        val featuredLink: Any? = null,
+        val createdAt: String? = null,
+        val deletedBy: Any? = null,
+        val topicTimer: Any? = null,
+        val title: String? = null,
+        val postStream: PostStream? = null,
+        val timelineLookup: List<List<Int?>?>? = null,
```

Comment:
Do we need all these params?
Let's only include those that we need, for example wordcount, pinnedAt, fancyTitle, slug, views, etc. aren't needed.

Same goes for other models too